.pipelines/templates/template-acr-push.yml

```diff
@@ -0,0 +1,73 @@
+parameters:
+  - name: jobName
+    type: string
+  - name: acrFQDN
+    type: string
+  - name: repository  # This is the repository in the ACR
+    type: string
+  - name: imageRepoName
+    type: string  # The local image repository name
+  - name: imageTag
+    type: string
+  - name: dockerFilePath
+    type: string
+  - name: pushLatest
+    type: boolean
+    default: false
+
+steps:
+  # Authenticate to ACR and push the image
```

Comment:
We should document in a comment why this template is even necessary (the answer is we want the ability to push images across Azure tenants, and MSI won't cut it) so that future maintainers understand why we didn't go the "easy" route of configuring `Docker@2`'s `push` command.

Reply:
I've added the comment explaining why this template is necessary for cross-tenant authentication.